typescript-carousel/main.ts

```diff
@@ -0,0 +1,195 @@
+const $img1 = document.querySelector('#img-1');
+if (!$img1) throw new Error('Could not load image 1');
+const $img2 = document.querySelector('#img-2');
+if (!$img2) throw new Error('Could not load image 2');
+const $img3 = document.querySelector('#img-3');
+if (!$img3) throw new Error('Could not load image 3');
+const $img4 = document.querySelector('#img-4');
+if (!$img4) throw new Error('Could not load image 4');
+const $img5 = document.querySelector('#img-5');
+if (!$img5) throw new Error('Could not load image 5');
+const $dot1 = document.querySelector('#dot1');
+if (!$dot1) throw new Error('Could not load dot 1');
+const $dot2 = document.querySelector('#dot2');
+if (!$dot2) throw new Error('Could not load dot 2');
+const $dot3 = document.querySelector('#dot3');
+if (!$dot3) throw new Error('Could not load dot 3');
+const $dot4 = document.querySelector('#dot4');
+if (!$dot4) throw new Error('Could not load dot 4');
+const $dot5 = document.querySelector('#dot5');
+if (!$dot5) throw new Error('Could not load dot 5');
+const $right = document.querySelector('#right');
+if (!$right) throw new Error("Couldn't load right arrow");
+const $left = document.querySelector('#left');
+if (!$left) throw new Error("Couldn't load left arrow");
+const $dots = document.querySelectorAll('span');
+if (!$dots) throw new Error("Couldn't load left arrow");
+// what
+let intervalId = setInterval(imgSwitch, 3000);
+
+function startInterval(): void {
+  clearInterval(intervalId);
+  intervalId = setInterval(imgSwitch, 3000);
+}
+function imgSwitch(): void {
+  if (!$img1) throw new Error('Could not load image 1');
+  if (!$img2) throw new Error('Could not load image 2');
+  if (!$img3) throw new Error('Could not load image 3');
+  if (!$img4) throw new Error('Could not load image 4');
+  if (!$img5) throw new Error('Could not load image 5');
+  if (!$dot1) throw new Error('Could not load dot 1');
+  if (!$dot2) throw new Error('Could not load dot 2');
+  if (!$dot3) throw new Error('Could not load dot 3');
+  if (!$dot4) throw new Error('Could not load dot 4');
+  if (!$dot5) throw new Error('Could not load dot 5');
```

Comment:
You have these checks in the global space, do you need these inside of the function?